src/java.base/share/classes/sun/nio/ch/UnixDomainSockets.java

```diff
@@ -137,6 +139,8 @@ static UnixDomainSocketAddress generateTempName() throws IOException {
             return UnixDomainSocketAddress.of(path);
         } catch (InvalidPathException e) {
             throw new BindException("Invalid temporary directory");
+        } catch (IllegalArgumentException e) {
+            throw new UnsupportedOperationException("Unix Domain Sockets not supported on non-default file system");
```

Comment:
There should be no need to catch IAE here, instead you can check the provider.

Reply:
Changed to `if (path.getFileSystem().provider() != sun.nio.fs.DefaultFileSystemProvider.instance()) ...`
I hope that was what you meant.